web/js/codeworld_shared.js

```diff
@@ -122,95 +122,142 @@ var hintBlacklist = [
     "Pi",
 ];
 
-var codeWorldBuiltinsDocs = {};
 
-function onHover(cm, data, node){
-    if (data && data.token && data.token.string) {
-        var token_name = data.token.string;
-        if (hintBlacklist.indexOf(token_name) == -1 && codeWorldBuiltinsDocs[token_name]){
-            return codeWorldBuiltinsDocs[token_name];
+// codeWorldDocs is variable containing annotations and documentation
+// of builtin and user-defined variables.
+// Expected format:
+// codeWorldDocs = {
+//   codeWorldLogo: {
+//     annotation: "Picture",
+//     doc: "The CodeWorld logo.",
+//     builtin: true // codeWorld definitions, not from current project
+//   }
+// }
+
+var codeWorldDocs = {};
+
+function codeToDoc() {
```

Comment:
This currently adds any new symbols defined in the source code, but it doesn't remove any old ones.  That's a problem.

Thinking about this, actually, I think the right thing to do is something like this:

    var codeWorldBuiltinDocs = {};
    var codeWorldDocs = {};

    function codeToDoc() {
      codeWorldDocs = Object.assign({}, codeWorldBuiltinDocs);
      ... // add symbols from the current module
    }

That way you start with a fresh copy of the builtins each time, and don't risk leaving around whatever weird stuff you might have picked up from the previous parse of the loaded code.

This way, you don't really need the `builtin` property.